### PR TITLE
Se/swerve

### PR DESCRIPTION
**Pull request by naowalrahman**
BOOM!

### CODE REVIEW

File: src/main/java/com/stuypulse/robot/commands/swerve/SwerveDriveDrive.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+            }
+            if () {
+
+            }
+        }
+
+        else {
+
+        }*/
```
Comment (BenG49) [inline src/main/java/com/stuypulse/robot/commands/swerve/SwerveDriveDrive.java]: finish

File: src/main/java/com/stuypulse/robot/constants/Settings.java [hunk 2]
```diff
@@ -16,4 +19,61 @@
  */
 public interface Settings {
 	double DT = 0.02;
+
+	public interface Swerve {
+        double WIDTH = Units.inchesToMeters(26.504);
+        double LENGTH = Units.inchesToMeters(20.508);
```
Comment (BenG49) [inline src/main/java/com/stuypulse/robot/constants/Settings.java]: ask harry what the drivetrain size is

File: src/main/java/com/stuypulse/robot/constants/Settings.java [hunk 3]
```diff
@@ -16,4 +19,61 @@
...
+
+		SmartNumber MODULE_SPEED_DEADBAND = new SmartNumber("Swerve/Module speed deadband", 0.02);
+		SmartNumber MAX_MODULE_SPEED = new SmartNumber("Swerve/Maximum module speed", 5);
```
Comment (BenG49) [inline src/main/java/com/stuypulse/robot/constants/Settings.java]: ask harry max theoretical speed

File: src/main/java/com/stuypulse/robot/constants/Settings.java [hunk 4]
```diff
@@ -16,4 +19,61 @@
...
+            SmartNumber kD = new SmartNumber("Swerve/Drive/kD", 0.0);
+
+            SmartNumber kS = new SmartNumber("Swerve/Drive/kV", 0.2230);
+            SmartNumber kV = new SmartNumber("Swerve/Drive/kV", 2.4899);
+            SmartNumber kA = new SmartNumber("Swerve/Drive/kA", 0.41763);
+        }
+		
+ 		public interface FrontRight {
+            String ID = "Front Right";
+            Rotation2d ABSOLUTE_OFFSET = Rotation2d.fromDegrees(357.213206) // recalibrated 4/21
```
Comment (BenG49) [inline src/main/java/com/stuypulse/robot/constants/Settings.java]: set all offsets to zero and remove comments for now

File: src/main/java/com/stuypulse/robot/subsystems/SL_SwerveModule.java
Comment (BenG49) [inline src/main/java/com/stuypulse/robot/subsystems/SL_SwerveModule.java:1]: rename to SwerveModuleImpl

File: src/main/java/com/stuypulse/robot/subsystems/SL_SwerveModule.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.SwerveModulePosition;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+
+public class SL_SwerveModule extends SwerveModule {
+    // data
+    private final String id;
+    private SwerveModuleState targetState;
+    private Translation2d translationOffset;
```
Comment (BenG49) [inline src/main/java/com/stuypulse/robot/subsystems/SL_SwerveModule.java]: this can also be final

**Review by BenG49 — CHANGES_REQUESTED** [review]
do this stuff

File: src/main/java/com/stuypulse/robot/subsystems/swerve/SwerveModule.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+public abstract class SwerveModule extends SubsystemBase {
+    public abstract String getID();
+    public abstract Translation2d getOffset();
+    public abstract SwerveModuleState getState();
+    public abstract SwerveModulePosition getModulePosition();
+
+    public abstract void setState(SwerveModuleState state);
+
+    @Override
+    public abstract void periodic();
```
Comment (BenG49) [inline src/main/java/com/stuypulse/robot/subsystems/swerve/SwerveModule.java]: whys this here, doesnt it extend subsystembase anyways

Comment (naowalrahman) [reply src/main/java/com/stuypulse/robot/subsystems/swerve/SwerveModule.java]: Probably an accident, it extended the wrong class and I didn't realize but when I fixed it I forgot to remove that

**Pull request merged**